test/Microsoft.AspNet.Mvc.Core.Test/ControllerTests.cs

```diff
@@ -374,6 +374,168 @@ public void RedirectToRoutePermanent_WithParameterRouteName_SetsResultSameRouteN
         }
 
         [Fact]
+        public void Created_WithStringParameter_SetsCreatedLocation()
+        {
+            // Arrange
+            var controller = new Controller();
+            var uri = "http://test/url";
+
+            // Act
+            var result = controller.Created(uri, null);
+
+            // Assert
+            Assert.IsType<CreatedResult>(result);
+            Assert.Equal(201, result.StatusCode);
+            Assert.Same(uri, result.Location);
+        }
+
+        [Fact]
+        public void Created_WithAbsoluteUriParameter_SetsCreatedLocation()
+        {
+            // Arrange
+            var controller = new Controller();
+            var uri = new Uri("http://test/url");
+
+            // Act
+            var result = controller.Created(uri, null);
+
+            // Assert
+            Assert.IsType<CreatedResult>(result);
+            Assert.Equal(201, result.StatusCode);
+            Assert.Same(uri.OriginalString, result.Location);
```

Comment:
Use `Assert.Equal` here. We don't want to depend on the behavior of `OriginalString` returning the same instance of string every time here for the `Assert.Same` to succeed.